fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -434,7 +434,12 @@ public Response updateSparql(@ContentLocation final InputStream requestBodyStrea
             }
             session.save();
 
-            addCacheControlHeaders(servletResponse, resource(), session);
+            if (resource() instanceof FedoraBinary) {
+                final NonRdfSourceDescription description = ((FedoraBinary)resource()).getDescription();
```

Comment:
@ajs6f are you suggesting that we move the `getDescription` method to `FedoraResource` as part of this PR or would that be something separate?